ewebsock/src/lib.rs

```diff
@@ -77,8 +77,8 @@ pub enum WsEvent {
     /// An error occurred.
     Error(String),
 
-    /// The connection has been closed.
-    Closed,
+    /// The connection has been closed, possibly with a reason code.
+    Closed(Option<u16>),
```

Comment:
We should add the human-readable "reason" here too from https://docs.rs/tungstenite/latest/tungstenite/protocol/frame/struct.CloseFrame.html#structfield.reason or https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent/reason

We should link to something that explains this code, e.g. https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent/code